File: frontend/src/types/notification.ts
export interface Notification {
  id: number;
  type: "digest" | "staleness" | "pattern";
  content: string;
  read: boolean;
  read_at?: string;
  suppressed_by?: string;
  completed: boolean;
  completed_at?: string;
  created_at: string;
}

File: backend/src/istari/models/notification.py
"""Notification model — worker queues these; API serves them to the frontend."""

import datetime

from sqlalchemy import Boolean, DateTime, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from istari.models.base import Base, TimestampMixin


class Notification(TimestampMixin, Base):
    __tablename__ = "notifications"

    id: Mapped[int] = mapped_column(primary_key=True)
    type: Mapped[str] = mapped_column(String(50))
    content: Mapped[str] = mapped_column(Text)
    read: Mapped[bool] = mapped_column(Boolean, default=False)
    read_at: Mapped[datetime.datetime | None] = mapped_column(DateTime(timezone=True))
    suppressed_by: Mapped[str | None] = mapped_column(String(50))
